core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -402,6 +404,15 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
       false
   }
 
+  def isUnknownCommit(memberId: String,
+                      producerId: Long): Boolean = {
+    if (producerId != NO_PRODUCER_ID) {
```

Comment:
Although it is possible to create a more condensed `xx and (yy || zz)` expression, I found current verbose version is more readable and understandable.

Reply:
prop: See my other comment: I think manual commit could not happen for txnal commit, and if yes we could make the caller logic more explicit -- it means a bit more LOC, but the logic is more readable.